dashboard/app/views/levels/_level_group.haml

```diff
@@ -77,12 +71,10 @@
         - unless @script_level.nil?
           = render partial: 'levels/dialog', locals: {app: app, data: data, previous_button: page.page_number > 1, next_button: page.page_number < @total_page_count}
 
-- unless @script_level.nil?
-  :javascript
-    $(document).ready(function() {
-      window.initLevelGroup(
-        #{@total_level_count},
-        #{current_page || -1},
-        #{@last_attempt || {}}
-      );
-    });
+- content_for(:head) do
+  - level_data = data
+  - init_data = nil
+  - unless @script_level.nil?
+    - init_data = { total_level_count: @total_level_count, page: @current_page || -1, last_attempt: parsed_last_attempt}
```

Comment:
I think this should be `current_page`, not `@current_page`? It's a local assigned here:
https://github.com/code-dot-org/code-dot-org/pull/18316/files#diff-5dae8e5ed37046fce3fd66a9f5ae6189R23

Reply:
Second thing - I think we used to pass the unparsed `@last_attempt` into `initLevelGroup`, now you're parsing it first.  Intentional?

Reply:
I think you're right for current_page. Good catch.

initLevelGroup expects an object (rather than a JSON string) as its third param. I just double checked, and that's what it was receiving before my changes. It's not entirely clear to me why, as I would expect `@last_attempt` to be a JSON string here. Perhaps some ruby magic with it being inside a :javascript block? I might poke around some more to see if I can understand.

Reply:
I see. As far as I can tell, `@last_attempt` is a string (in Ruby) containing an object in JSON format (we do pass it to `JSON.parse` earlier). Inside the HAML `:javascript` block we were inlining that string, which would just look like an inlined JS object in the rendered template.

`parsed_last_attempt` will be a Ruby hash though (or something like it?) - but I guess that's fine, since you add it to a hash that then runs through `to_json` below.  I think could only introduce a problem if the serialization round-trip isn't stable for the last attempt data, so probably not an issue.

Reply:
I think I was struggling with why it wasn't json inside the `:javascript` block, but I've now convinced myself this makes sense.

I had initially been passing this as a JSON string, but it felt weird to end up calling JSON.parse twice on the client (once for the `initData` object, then a second time for `initData.lastAttempt`).